app/github_repo_fetcher.rb

```diff
@@ -32,13 +32,22 @@ def docs(app_name)
           title: title,
           markdown: contents,
           source_url: doc.html_url,
+          latest_commit: latest_commit(app_name, doc.path),
         }
       end
     rescue Octokit::NotFound
       nil
     end
   end
 
+  def latest_commit(app_name, path)
+    latest_commit = client.commits("alphagov/#{app_name}", repo(app_name).default_branch, path: path).first
```

Comment:
I did try that initially, but `sha` doesn't give what we expect. Take `https://github.com/alphagov/email-alert-api/blob/master/docs/analytics.md` for example - this is what we get:

```ruby
{:name=>"analytics.md",
 :path=>"docs/analytics.md",
 :sha=>"1582026ed9f57dda0b83d0b25656f3313e88f5e3",
 :size=>6007,
 :url=>
  "https://api.github.com/repos/alphagov/email-alert-api/contents/docs/analytics.md?ref=master",
 ...
```

The `sha` gives a 404: https://github.com/alphagov/email-alert-api/commit/1582026ed9f57dda0b83d0b25656f3313e88f5e3

I think the `sha` here is actually the [blob SHA of the file](https://developer.github.com/v3/repos/contents/#parameters-2), rather than the commit hash. Useful for sanitation purposes, but not useful as a git reference!

So unfortunately I had to make an additional call to the API, which gives us the right SHA value:

```ruby
latest_commit(app_name, doc.path)

 {:sha=>"1a25d44126709bf8a98955d7609ced439095dd9c", :timestamp=>2020-09-09 08:54:40 UTC}
```

Reply:
Ah, that's a shame. Thanks for checking: I didn't appreciate the SHAs were specific to the API resource ("blob" vs. "commit"). I agree this seems to be necessary to get the data we need: because of the way Git works, we need to ask about each file individually in order to know when it was last committed i.e. modified.

I did find one way we could speed this up, though: can we add a `per_page` limit of 1?

Reply:
I like the suggestion, unfortunately it [seems to end up making a lot more requests](https://github.com/alphagov/govuk-developer-docs/pull/2784#discussion_r509438676). (For the record, I was adding a `per_page: 1` to the `client.commits` call in GitHubRepoFetcher.